File: backend/src/main/java/com/dinebook/backend/dto/CreateDiningRequest.java
package com.dinebook.backend.dto;

import jakarta.validation.constraints.Future;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotNull;

import java.time.LocalDateTime;

public record CreateDiningRequest(
        @NotNull Long restaurantId,
        @NotNull @Future LocalDateTime requestedDateTime,
        @Min(1) int guests
) {}

File: mobile/app/src/main/java/com/dinebook/mobile/booking/CreateDiningRequest.kt
package com.dinebook.mobile.booking

import com.google.gson.annotations.SerializedName

data class CreateDiningRequest(
    @SerializedName("restaurantId")
    val restaurantId: Long,
    @SerializedName("requestedDateTime")
    val requestedDateTime: String,
    val guests: Int
)
